### ISSUE TITLE
RouteTable.Add with DI injected presenter

### ISSUE THREAD

**Issue by exyi** [riganti/dotvvm#463]
The `RouteTable.Add` method has a `presenterFactory` parameter, that can be used for creating a custom presenter, but it can't easily come from DI as there is no `IServiceProvider` parameter. I think that we should add an overload that would allow that. The presenter injection could then look like:
```C#
routes.Add(..., presenterFactory: ServiceProviderServiceExtensions.GetRequiredService<SomePresenter>)
```

Do you think that it's enough, or one more overload with something like `presenterType: Type` would be useful?

**Comment by djanosik** [comment]
The solution with `presenterFactory` overload feels good enough. We can add `presenterType` later if needed.

**Comment by tomasherceg** [comment]
I think that having the overload with just presenterType is a nice way - also, the custom presenter doesn't need the view path parameter so we can get rid of this parameter.

**Comment by exyi** [comment]
I'd only implement the overload with `presenterFactory`, it's more explicit, not that much verbose and this feature is probably not going to be used much in real projects, so readability is more important than writability :).

**Comment by Leeemons** [comment]
https://github.com/riganti/dotvvm/commit/644cd589932996dc9f05ef6e0800a5e275544da3

**Comment by exyi** [comment]
This does not solve the issue at all, we wanted an overload with the `presenterFactory` factory of type `Func<IServiceProvider, IDotvvmPresenter>` and we were uncertain about the generic form (I don't like it very much...)

**Comment by Leeemons** [comment]
I created PR #514

**Comment by quigamdev** [comment]
Solved by #514
